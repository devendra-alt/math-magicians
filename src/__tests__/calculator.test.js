import '@testing-library/jest-dom';
import { MemoryRouter } from 'react-router-dom';
import renderer from 'react-test-renderer';
import React from 'react';
import Calculator from '../components/calculator.js';
import { fireEvent, render, screen } from '@testing-library/react';
import userEvent from '@testing-library/user-event';
describe('renders correctly', () => {
  beforeEach(() => {
    render(
      <MemoryRouter>
        <Calculator />
      </MemoryRouter>
    );
  });
  it('renders correctly', () => {
    const tree = renderer
      .create(
        <MemoryRouter>
          <Calculator />
        </MemoryRouter>
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
  it('should have calculator have title on screen', async () => {
    const titleNode = await screen.getByText("Let's do some math");
    expect(titleNode).toHaveTextContent("Let's do some math");
  });
  it('should have calculator buttons on screen', () => {
    const buttonNodes = screen.getAllByLabelText('calculator-btn');
    expect(buttonNodes).toHaveLength(19);
  });
  it('should add running value (aka next)', () => {
    const buttonNodes = screen.getAllByLabelText('calculator-btn');
    let sevenNode = null;
    let fiveNode = null;
    buttonNodes.forEach((element) => {
      if (element.textContent === '7') {
        sevenNode = element;
      }
      if (element.textContent === '5') {
        fiveNode = element;
      }
    });

    sevenNode.innerText = '7';
    fiveNode.innerText = '5';

    fireEvent.click(sevenNode);
    fireEvent.click(fiveNode);

    const answer = screen.getByLabelText('calculator-answer-p-span-next');
    expect(answer.textContent).toBe('75');
  });
  it('Backspace should work', async () => {
    const answer_ = screen.getByLabelText('calculator-answer-p-span-next');
    await userEvent.keyboard('{7}');
    await userEvent.keyboard('{7}');
    await userEvent.keyboard('{7}');
    await userEvent.keyboard('{7}');
    await userEvent.keyboard('{Backspace}');
    expect(answer_.textContent).toBe('777');
  });
});
