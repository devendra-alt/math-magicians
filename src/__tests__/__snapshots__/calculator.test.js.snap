// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly renders correctly 1`] = `
Array [
  <header
    className="header"
  >
    <nav
      className="nav-bar"
    >
      <h1
        className="title"
      >
        Math Magicians
      </h1>
      <ul
        className="nav-links"
      >
        <li
          className="nav-link-item"
        >
          <a
            className="nav-link"
            href="/"
            onClick={[Function]}
          >
            Home
          </a>
        </li>
        <li
          className="nav-link-item"
        >
          <a
            className="nav-link"
            href="/calculator"
            onClick={[Function]}
          >
            Calculator
          </a>
        </li>
        <li
          className="nav-link-item"
        >
          <a
            className="nav-link"
            href="/quotes"
            onClick={[Function]}
          >
            Quotes
          </a>
        </li>
      </ul>
    </nav>
  </header>,
  <div
    className="calculator-container"
  >
    <p
      className="calculator-text"
    >
      Let's do some math
    </p>
    <article>
      <table>
        <tbody>
          <tr>
            <td
              colSpan={4}
              id="calculator-answer-el"
            >
              <p
                aria-label="calculator-answer-p-el"
              >
                <span
                  aria-label="calculator-answer-p-span-result"
                />
                <span
                  aria-label="calculator-answer-p-span"
                />
                <span
                  aria-label="calculator-answer-p-span-next"
                >
                  0
                </span>
              </p>
            </td>
          </tr>
          <tr>
            <td
              aria-label="calculator-btn"
              onClick={[Function]}
            >
              AC
            </td>
            <td
              aria-label="calculator-btn"
              onClick={[Function]}
            >
              +/-
            </td>
            <td
              aria-label="calculator-btn"
              onClick={[Function]}
            >
              %
            </td>
            <td
              aria-label="calculator-btn"
              onClick={[Function]}
            >
              รท
            </td>
          </tr>
          <tr>
            <td
              aria-label="calculator-btn"
              onClick={[Function]}
            >
              7
            </td>
            <td
              aria-label="calculator-btn"
              onClick={[Function]}
            >
              8
            </td>
            <td
              aria-label="calculator-btn"
              onClick={[Function]}
            >
              9
            </td>
            <td
              aria-label="calculator-btn"
              onClick={[Function]}
            >
              x
            </td>
          </tr>
          <tr>
            <td
              aria-label="calculator-btn"
              onClick={[Function]}
            >
              4
            </td>
            <td
              aria-label="calculator-btn"
              onClick={[Function]}
            >
              5
            </td>
            <td
              aria-label="calculator-btn"
              onClick={[Function]}
            >
              6
            </td>
            <td
              aria-label="calculator-btn"
              onClick={[Function]}
            >
              -
            </td>
          </tr>
          <tr>
            <td
              aria-label="calculator-btn"
              onClick={[Function]}
            >
              1
            </td>
            <td
              aria-label="calculator-btn"
              onClick={[Function]}
            >
              2
            </td>
            <td
              aria-label="calculator-btn"
              onClick={[Function]}
            >
              3
            </td>
            <td
              aria-label="calculator-btn"
              onClick={[Function]}
            >
              +
            </td>
          </tr>
          <tr>
            <td
              aria-label="calculator-btn"
              className="calculator-zero-el"
              colSpan={2}
              onClick={[Function]}
            >
              0
            </td>
            <td
              aria-label="calculator-btn"
              onClick={[Function]}
            >
              .
            </td>
            <td
              aria-label="calculator-btn"
              onClick={[Function]}
            >
              =
            </td>
          </tr>
        </tbody>
      </table>
    </article>
  </div>,
]
`;
